### ISSUE TITLE
Quest villagers don't trade with me after I finished his questline

### ISSUE THREAD

**Issue by InteJason** [SaphieNyako/Quest_Giver#16]
Title, I tried this with the signed trading scrolls, it worked on regular villagers only.

Expected Behavior: After finishing the quest villager's questline, they will lose their exclamation/question mark, and perform regular trading function.

**Comment by SaphieNyako** [comment]
It's not after the questline is finish, its because you recieve a trading contract, if you offer the trading contract you can trade. 

I didnt add the discount. 

**Behaviour after fix:** After Finishing the questline, you receive a trading contract, when signed you can offer this contract to a NPC with that profession. It will now open a trading window and give discount. 


![2022-08-07_17 06 22](https://user-images.githubusercontent.com/77398437/183297596-e93fed6b-e3b3-4c04-bef4-0a1abeaa402e.png)

**Comment by InteJason** [comment]
I can't trade with quest villager still, even with the trading scroll. In the latest version you sent me, I can't trade with villagers as well, while I was able to do so in previous versions.

How to replicate this issue:
1) Get trading scroll of any guild using /give command
2) Sign it using crafting table
3) Go to a regular villager/quest villager(After finishing their questline) and right click on them using the signed trading scroll
4) The villager doesn't trade.

It could be because of other mods conflicting, i've turned off the reputation mod's trading restriction to test if its the cause, it isn't.

However, I did notice that there's an error with the trading scroll in the log. Don't know if it means anything: https://paste.ee/p/c1NaN
> java.lang.NullPointerException: Cannot invoke "String.equals(Object)" because the return value of "com.feywild.quest_giver.item.TradingContract.playerSignature()" is null
